### PR TITLE
Display road closures on traffic line.

### LINKED ISSUE #4188: Road closure styling option needs to be added
Recently the directions API started returning road closure data in the directions response when specific options are provided. 
The sections of a route that are closed need to be visible on the traffic line along with the traffic congestion that is currently represented.

There needs to be options to configure the color for closed sections and closed sections should be represented along with the traffic congestion on the traffic line.

A reference from the directions API docs: https://user-images.githubusercontent.com/8186438/108923695-24808200-75ee-11eb-8c50-fdfa98b058a1.png

This issue is dependent on this issue: https://github.com/mapbox/mapbox-java/issues/1249

### PR DESCRIPTION
**Pull request by cafesilencio**
### Description
Resolves #4188 by displaying road closure(s) on the route traffic line.

In order to test this feature you need a route with a road closure as well as the following RouteOptions when requesting a route:

profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC
snappingClosures ( listOf(true, true) ) // makes it easier to create a route starting in a closure, normally the directions API will route around closures rather than going through them.
annotationsList( listOf("closure", "congestion", "distance") )

### Changelog
```
<changelog>Added option to display road closures on the route traffic line.</changelog>
```

### Screenshots or Gifs
In the screen shot below the color options were configured to show low traffic congestion as yellow and road closures as black.
Screenshot_20210331-090535

### CODE REVIEW

File: libnavui-base/src/main/java/com/mapbox/navigation/ui/base/internal/model/route/RouteConstants.kt [hunk 1]
```diff
@@ -110,6 +111,12 @@ object RouteConstants {
     @ColorInt
     val MANEUVER_ARROW_BORDER_COLOR = Color.parseColor("#2D3F53")
 
+    @ColorInt
+    val ROUTE_CLOSURE_COLOR = Color.parseColor("#000000")
```
Comment (Guardiola31337) [inline libnavui-base/src/main/java/com/mapbox/navigation/ui/base/internal/model/route/RouteConstants.kt:115]: Do we want to default to `#000000`? 🤔 

cc @d-prukop

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/RouteLineColorResources.kt [hunk 2]
```diff
@@ -95,7 +99,10 @@ class RouteLineColorResources private constructor(
             "alternativeRouteModerateColor=$alternativeRouteModerateColor" +
             "alternativeRouteHeavyColor=$alternativeRouteHeavyColor" +
             "alternativeRouteSevereColor=$alternativeRouteSevereColor" +
-            "alternativeRouteCasingColor=$alternativeRouteCasingColor"
+            "alternativeRouteCasingColor=$alternativeRouteCasingColor" +
+            "routeClosureColor=$routeClosureColor" +
+            "alternativeRouteClosureColor=$alternativeRouteClosureColor" +
```
Comment (Guardiola31337) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/RouteLineColorResources.kt:104]: This also applies to the rest of the fields.

```suggestion
            "alternativeRouteCasingColor=$alternativeRouteCasingColor, " +
            "routeClosureColor=$routeClosureColor, " +
            "alternativeRouteClosureColor=$alternativeRouteClosureColor, " +
```

File: libnavui-base/src/main/java/com/mapbox/navigation/ui/base/internal/model/route/RouteConstants.kt (hunk 1 above)
Comment (Guardiola31337) [reply libnavui-base/src/main/java/com/mapbox/navigation/ui/base/internal/model/route/RouteConstants.kt:115]: We can follow up in a separate ticket. Could you cut a ticket when you have a chance @cafesilencio? 🙏

**Review by Guardiola31337 — APPROVED** [review]

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/RouteLineColorResources.kt (hunk 2 above)
Comment (cafesilencio) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/RouteLineColorResources.kt:104]: done

**Pull request merged**